lib/spack/spack/spec.py

```diff
@@ -1401,15 +1422,16 @@ def _add_flag(self, name, value):
         elif name in valid_flags:
             assert(self.compiler_flags is not None)
             self.compiler_flags[name] = spack.compiler.tokenize_flags(value)
+            self.compiler_flags.propagate = propagate
```

Comment:
```suggestion
            self.compiler_flags.propagate[name] = propagate
```
The `FlagMap` class would also need to be updated to handle this.